src/main/java/org/myrobotlab/i2c/I2CBus.java

```diff
@@ -133,4 +137,61 @@ public boolean isLocal() {
     return true;
   }
 
+  @Override
+  public boolean hasInterface(Class<?> class1) {
+    Class<?>[] faces = I2CBus.class.getInterfaces();
+    for (Class<?> c : faces) {
+      if (c.equals(class1)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  @Override
+  public boolean hasInterface(String inter) {
+    Class<?>[] interfaces = getClass().getInterfaces();
+    for (int i = 0; i < interfaces.length; ++i) {
+      Class<?> interfaze = interfaces[i];
+      if (inter.getClass().getCanonicalName().equals(inter)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  @Override
+  public boolean isType(Class<?> clazz) {
+    return isType(clazz.getCanonicalName());
+  }
+
+  @Override
+  public boolean isType(String clazz) {
+    return getClass().getCanonicalName().equals(clazz);
+  }
+
+  @Override

```

Comment:
remove these methods. they don't do anything.

Reply:
Same as TimeEncoder this should have been a service